primitives/manta/src/assets.rs

```diff
@@ -14,178 +14,237 @@
+        AccountId = C::AccountId,
+        AssetId = Self::AssetId,
+        Balance = Self::Balance,
+    >;
 }
 
-/// The metadata of a Manta Asset
-#[derive(Clone, Eq, Debug, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
-pub struct AssetRegistrarMetadata {
+/// Asset Storage Metadata
+#[derive(Clone, Debug, Decode, Encode, Eq, Hash, Ord, PartialEq, PartialOrd, TypeInfo)]
+pub struct AssetStorageMetadata {
+    /// Asset Name
     pub name: Vec<u8>,
+
+    /// Asset Symbol
     pub symbol: Vec<u8>,
+
+    /// Number of Decimals
     pub decimals: u8,
-    pub evm_address: Option<H160>,
+
+    /// Frozen Flag
+    ///
+    /// Whether or not transfers of the asset are allowed.
     pub is_frozen: bool,
-    pub min_balance: Balance,
-    /// `is_sufficient`: Whether a non-zero balance of this asset is deposit of sufficient
-    /// value to account for the state bloat associated with its balance storage. If set to
-    /// `true`, then non-zero balances may be stored without a `consumer` reference (and thus
-    /// an ED in the Balances pallet or whatever else is used to control user-account state
-    /// growth).
-    /// For example, if is_sufficient set to `false`, a fresh account cannot receive XCM tokens.
+}
+
+impl<B> From<AssetRegistryMetadata<B>> for AssetStorageMetadata {
+    #[inline]
+    fn from(source: AssetRegistryMetadata<B>) -> Self {
+        source.metadata
+    }
+}
+
+/// Asset Registry Metadata
+#[derive(Clone, Debug, Decode, Encode, Eq, Hash, Ord, PartialEq, PartialOrd, TypeInfo)]
+pub struct AssetRegistryMetadata<B> {
+    /// Asset Storage Metadata
+    pub metadata: AssetStorageMetadata,
+
+    /// Optional EVM Address for the Asset
+    pub evm_address: Option<H160>,
+
+    /// Minimum Balance
+    pub min_balance: B,
+
+    /// Sufficiency Flag
+    ///
+    /// This flag should be set to `true` whenever a non-zero balance of this asset is deposit of
+    /// sufficient value to account for the state bloat associated with its balance storage. If set
+    /// to `true`, then non-zero balances may be stored without a `consumer` reference (and thus an
+    /// existential deposit in the balances pallet or whatever else is used to control user-account
+    /// state growth).
+    ///
+    /// If this flag is set to `false`, a fresh account cannot receive XCM tokens.
     pub is_sufficient: bool,
 }
 
-impl Default for AssetRegistrarMetadata {
+impl Default for AssetRegistryMetadata<MantaBalance> {
     fn default() -> Self {
         Self {
-            name: b"Dolphin".to_vec(),
-            symbol: b"DOL".to_vec(),
-            decimals: 12,
+            metadata: AssetStorageMetadata {
+                name: b"Default".to_vec(),
+                symbol: b"DEF".to_vec(),
+                decimals: 12,
+                is_frozen: false,
+            },
             evm_address: None,
-            is_frozen: false,
-            min_balance: DEFAULT_ASSET_ED,
+            min_balance: TEST_DEFAULT_ASSET_ED,
```

Comment:
For me, if I use `AssetRegistryMetadata::default()` to create an instance, I don't know it's for testing.

Reply:
Right, I see how this can be confusing.
How about I make a trait with a more explicit name like `DefaultForTesting`

Reply:
Ok I see how this can be confusing, especially since there's no such thing as a default asset metadata.
But this is just an easy convenient way to get a function that returns some asset that is of the correct type for testing/benchmarking in multiple modules.
I can add some comments for this or if you have a proper solution please let me know.

Reply:
Nvm I think I can just make another trait with better name `TestingDefault`